error-prone-contrib/src/main/java/tech/picnic/errorprone/refasterrules/ImmutableEnumSetRules.java

```diff
@@ -0,0 +1,177 @@
+package tech.picnic.errorprone.refasterrules;
+
+import com.google.common.collect.ImmutableSet;
+import com.google.common.collect.Sets;
+import com.google.errorprone.refaster.Refaster;
+import com.google.errorprone.refaster.annotation.AfterTemplate;
+import com.google.errorprone.refaster.annotation.BeforeTemplate;
+import com.google.errorprone.refaster.annotation.Repeated;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.EnumSet;
+import tech.picnic.errorprone.refaster.annotation.OnlineDocumentation;
+
+/** Refaster rules related to expressions dealing with immutable {@link Enum} collections. */
+@OnlineDocumentation
+final class ImmutableEnumSetRules {
+  private ImmutableEnumSetRules() {}
+
+  /**
+   * Prefer {@link Sets#immutableEnumSet(Iterable)} for enum collections to take advantage of
+   * internally used {@link EnumSet}.
+   */
+  static final class ImmutableEnumSetIterable<T extends Enum<T>> {
```

Comment:
The yet-to-be formalized Refaster naming algorithm would add the `Sets` here in front :).